src/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -269,6 +271,19 @@ public SamReader open(final SamInputResource resource) {
                             bufferedStream.close();
                             primitiveSamReader = new CRAMFileReader(sourceFile, indexFile, referenceSource, validationStringency);
                         }
+                    } else if (SamStreams.isSRAFile(bufferedStream)) {
+                        if (sourceFile == null || !sourceFile.isFile()) {
+                            sourceFile = null;
+                        } else {
+                            bufferedStream.close();
+                            bufferedStream = null;
+                        }
```

Comment:
Since you're not using the `bufferedStream` after the `SamStreams.isSRAFile(bufferedStream)` call, I think you want to close it unconditionally here.